docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,787 @@
+```
+Set this policy as "rules" into NetworkFilter.
+
+4. Shadow rules. Take only rules that have "DENY_WITH_SHADOW_ALLOW":
+
+```yaml
+rules:
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: !dev
+        kuma.io/service: !web
+    default:
+      action: DENY_WITH_SHADOW_ALLOW
+```
+
+Generate [config.rbac.v3.RBAC](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/rbac/v3/rbac.proto#envoy-v3-api-msg-config-rbac-v3-rbac)
+with "ALLOW" action and with a single "MeshTrafficPermission" policy. The policy has permission "any: true" and the list of "principals".
+List of principals generated based on the list from step 3. Each principal is matched with OR semantic:
+
+```yaml
+action: ALLOW
+policies:
+  "ShadowMeshTrafficPermission":
+    permissions:
+      - any: true
+    principals:
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://kuma.io/service/web"
+```
+Set this policy as "shadow_rules" into NetworkFilter.
+
+5. If MeshTrafficPermission is targeting proxy then NetworkRBAC is placed as a NetworkFilter on the inbound listener.
+If MeshTrafficPermission is targeting MeshHTTPRoute then NetworkRBAC is placed as a NetworkFilter on the route. 
+Even in the case of MeshHTTPRoute, global per-proxy NetworkRBAC will be set to provide secure access for other routes. 
+
+### MeshTrafficPermission and Kubernetes
+
+MeshTrafficPermission is going to be the first policy with a new policy matching. 
+There are a few things worth mentioning regarding this fact. 
+
+#### Mesh as label
+
+Instead of creating a CRD with a "mesh" field, we can use a label to store mesh:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrafficPermission
+metadata:
+  labels:
+    kuma.io/mesh: mesh-1
+```
+
+This allows filtering policies by mesh using kubectl. 
+If the label is unset it will be added automatically by the webhook to be "kuma.io/mesh: default".
+
+#### Scope is namespaced
+
+Today all policies are cluster-scoped, but new policies should be namespace scoped.
+MeshTrafficPermission can only be created in the namespace where the control-plane runs (`kuma-system` by default), 
+and affects DPPs across all namespaces.
+In the future we'll add support for other namespaces, so the policy could affect DPPs only in the specified namespace but this is out of scope for this MADR. 
+
+### Cross-mesh
+
+MeshTrafficPermission is used to specify a list of clients from other meshes that have an access to MeshGatewayRoute.
+In that case we use targetRef with "mesh" field:
+
+```yaml
+type: MeshTrafficPermission
+mesh: expose
```

Comment:
Do we need a mesh field here? Based on section `Mesh as label` I think we should get rid of it here.

Reply:
This is a `Universal` format, `Mesh as label` is only for k8s